### PR TITLE
Documenter 1.0 upgrade

### PR DESCRIPTION
**Pull request by ArnoStrouwen**

### CODE REVIEW

**Comment by ArnoStrouwen** [comment]
SDE example is only compatible with SciMLBase 2.
But not all other dependencies have been upgraded to version 2 yet.
BoundaryValueDiffEq, specifically.

There is also another error regarding a solver name that I don't know how to fix: GeneralMIRK4

**Comment by ArnoStrouwen** [comment]
Finally found the setting to get the buildkite working on PR.

**Comment by ChrisRackauckas** [comment]
https://github.com/SciML/BoundaryValueDiffEq.jl/pull/109 should be the only one left, and I hope it merges today.

File: docs/src/tutorials/bvp_example.md [hunk 1]
```diff
@@ -84,13 +84,14 @@ plot(sol3)
 Defining a similar problem as `TwoPointBVProblem` is shown in the following example.
 
 ```@example bvp
-function bc2!(residual, u, p, t) # u[1] is the beginning of the time span, and u[end] is the ending
-    residual[1] = u[1][1] + pi / 2 # the solution at the beginning of the time span should be -pi/2
-    residual[2] = u[end][1] - pi / 2 # the solution at the end of the time span should be pi/2
+function bc2!((resid_a, resid_b), (u_a, u_b), p, t) # u[1] is the beginning of the time span, and u[end] is the ending
```
Comment (ChrisRackauckas) [inline docs/src/tutorials/bvp_example.md:87]: @avik-pal is this not right?

Comment (avik-pal) [reply docs/src/tutorials/bvp_example.md:87]: The `t` should not be present

Comment (avik-pal) [inline docs/src/tutorials/bvp_example.md:87]:
```suggestion
function bc2!((resid_a, resid_b), (u_a, u_b), p) # u[1] is the beginning of the time span, and u[end] is the ending
```

**Review by avik-pal — CHANGES_REQUESTED** [review]

**Comment by ChrisRackauckas** [comment]
Looks like there's still more. @avik-pal can you complete this? Let's make sure that the constructors are "complete" before being set in stone.

File: docs/Project.toml [hunk 2]
```diff
@@ -39,7 +40,7 @@ AlgebraicMultigrid = "0.5, 0.6"
 BSON = "0.3"
 BVProblemLibrary = "0.1.2"
 BenchmarkTools = "1"
-BoundaryValueDiffEq = "4"
+BoundaryValueDiffEq = "5"
```
Comment (ChrisRackauckas) [inline docs/Project.toml:43]:
```suggestion
BoundaryValueDiffEq = "5.1"
```

File: docs/src/tutorials/bvp_example.md [hunk 3]
```diff
@@ -81,16 +81,27 @@ plot(sol3)
...
+Defining a similar problem as `TwoPointBVProblem` is shown in the following example:
 
 ```@example bvp
-function bc2!(residual, u, p, t) # u[1] is the beginning of the time span, and u[end] is the ending
-    residual[1] = u[1][1] + pi / 2 # the solution at the beginning of the time span should be -pi/2
-    residual[2] = u[end][1] - pi / 2 # the solution at the end of the time span should be pi/2
+function bc2a!(resid_a, u_a, p) # u_a is at the beginning of the time span
+    resid_a[1] = u_a[1] + pi / 2 # the solution at the beginning of the time span should be -pi/2
+end
+function bc2b!(resid_a, u_b, p) # u_b is at the ending of the time span
```
Comment (ChrisRackauckas) [inline docs/src/tutorials/bvp_example.md:93]:
```suggestion
function bc2b!(resid_b, u_b, p) # u_b is at the ending of the time span
```

**Pull request merged**